src/test/java/hudson/plugins/octopusdeploy/E2eTest.java

```diff
@@ -0,0 +1,99 @@
+package hudson.plugins.octopusdeploy;
+
+import com.octopus.sdk.Repository;
+import com.octopus.sdk.domain.BuildInformation;
+import com.octopus.sdk.domain.Space;
+import com.octopus.sdk.http.OctopusClient;
+import com.octopus.testsupport.OctopusDeployServerFactory;
+import hudson.model.FreeStyleProject;
+import okhttp3.OkHttpClient;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.SingleFileSCM;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/*
+E2E Tests run under JUnit-4 and test must be located in the
+default Maven test path due to Jenkins test harness limitations.
+ */
+public class E2eTest {
+
+    private final com.octopus.testsupport.OctopusDeployServer server = OctopusDeployServerFactory.create();
+    private Space space;
+
+    @Rule
+    public final JenkinsRule jenkinsRule = new JenkinsRule();
+
+    @Rule
+    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Before
+    public void setUp() throws IOException {
+        // Configure Jenkins Octopus Server connection
+        final OctopusDeployServer testServer = new OctopusDeployServer("TestServer",
+                server.getOctopusUrl(), server.getApiKey(), true);
+        final OctopusDeployPlugin.DescriptorImpl configDescriptor
+                = jenkinsRule.jenkins.getDescriptorByType(OctopusDeployPlugin.DescriptorImpl.class);
+        configDescriptor.setOctopusDeployServers(Collections.singletonList(testServer));
+        configDescriptor.save();
+
+        // Configure Jenkins OctoCLI
+        final OctoInstallation installation =
+                new OctoInstallation("Default", System.getenv("OCTOPUS_CLI_PATH"));
+        final OctoInstallation.DescriptorImpl cliDescriptor =
+                jenkinsRule.jenkins.getDescriptorByType(OctoInstallation.DescriptorImpl.class);
+        cliDescriptor.setInstallations(installation);
+        cliDescriptor.save();
+
+        // Get Space and configure Octopus server for tests
+        final OctopusClient client =
```

Comment:
nit: could use the OctopusClientFactory (which takes a ConnectData()) - but the code is probably similarly complex to this.
Having said that -at some stage, w're going to want something which converts from a jenkins::OctopusDeployServer to a ConnectData (just, maybe not now - this is just musing).